java-checks/src/main/java/org/sonar/java/checks/StringConcatToTextBlockCheck.java

```diff
@@ -19,29 +19,31 @@
  */
 package org.sonar.java.checks;
 
-import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
-import java.util.stream.Stream;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import org.sonar.check.Rule;
 import org.sonar.java.JavaVersionAwareVisitor;
 import org.sonar.java.model.ExpressionUtils;
+import org.sonar.java.model.LiteralUtils;
 import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.JavaFileScannerContext;
 import org.sonar.plugins.java.api.JavaVersion;
 import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
-import org.sonar.plugins.java.api.tree.ExpressionTree;
 import org.sonar.plugins.java.api.tree.LiteralTree;
 import org.sonar.plugins.java.api.tree.Tree;
 
 @Rule(key = "S6126")
 public class StringConcatToTextBlockCheck extends IssuableSubscriptionVisitor implements JavaVersionAwareVisitor {
 
   private static final String MESSAGE = "Replace this String concatenation with Text block.";
-  public static final int MINIMAL_LINE_LENGTH = 7;
+  public static final int MINIMAL_CONTENT_LENGTH = 19;
   public static final int MINIMAL_NUMBER_OF_LINES = 2;
+  // matches '\n' characters, but skips '\\n'
+  public static final Pattern EOL = Pattern.compile("(?<!\\\\)\\\\n");
```

Comment:
:smile: I didn't dare to propose regular expression lookbehind previously because I was unsure that tmembers of our java bubble team would be comfortable with it. But with a comment, it seems easy to understand :+1: